pypxe/tftp.py

```diff
@@ -121,39 +147,61 @@ def read(self, address, message):
             response += str(filesize)
             response += chr(0)
 
-        socknew = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        socknew.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        socknew.bind((self.ip, 0))
-        self.ongoing[address]['sock'] = socknew
+        # Create the data socket only if needed
+        if self.ongoing[address]['sock'] == None:
```

Comment:
This function (`.read()`) is only called once per client. Under what conditions has the socket already been created?